components/eam/src/physics/rrtmgp/radiation.F90

```diff
@@ -2744,6 +2792,7 @@ subroutine get_gas_vmr(icall, state, pbuf, gas_names, gas_vmr)
 
       ! initialize
       gas_vmr(:,:,:) = 0._r8
+      if (present(write_mmr).and.write_mmr) allocate(gas_out(pcols,pver))
```

Comment:
http://www.cs.rpi.edu/~szymansk/OOF90/bugs.html#1

See "Danger with Optional Arguments", this could potentially cause an out of bounds memory reference.

Reply:
ahhh good catch! Thanks